plugin/storage/s3/s3.go

```diff
@@ -0,0 +1,67 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	awss3 "github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/usememos/memos/api"
+)
+
+type Client struct {
+	Client     *awss3.Client
+	BucketName string
+	URLPrefix  string
+}
+
+func NewClient(ctx context.Context, storage *api.Storage) (*Client, error) {
+	r2Resolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+		return aws.Endpoint{
+			URL:           storage.EndPoint,
+			SigningRegion: storage.Region,
+		}, nil
+	})
+
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithEndpointResolverWithOptions(r2Resolver),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(storage.AccessKey, storage.SecretKey, "")),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	client := awss3.NewFromConfig(cfg)
+
+	return &Client{
+		Client:     client,
+		BucketName: storage.Bucket,
+		URLPrefix:  storage.URLPrefix,
+	}, nil
+}
+
+func UploadFile(ctx context.Context, client *Client, filename string, fileType string, src io.Reader, storage *api.Storage) (*string, error) {
```

Comment:
`UploadFile` should be a method of `Client`.

```suggestion
func (client *Client) UploadFile(ctx context.Context, filename string, fileType string, src io.Reader, storage *api.Storage) (*string, error) {
```